beacon-chain/core/state/transition.go

```diff
@@ -404,41 +420,46 @@ func ProcessBlockNoVerifyAnySig(
 	ctx, span := trace.StartSpan(ctx, "beacon-chain.ChainService.state.ProcessBlockNoVerifyAnySig")
 	defer span.End()
 
-	state, err := b.ProcessBlockHeaderNoVerify(state, signed.Block)
-	if err != nil {
-		traceutil.AnnotateError(span, err)
-		return nil, nil, errors.Wrap(err, "could not process block header")
-	}
-	bSet, err := b.BlockSignatureSet(state, signed)
-	if err != nil {
-		traceutil.AnnotateError(span, err)
-		return nil, nil, errors.Wrap(err, "could not retrieve block signature set")
-	}
-	rSet, err := b.RandaoSignatureSet(state, signed.Block.Body)
-	if err != nil {
-		traceutil.AnnotateError(span, err)
-		return nil, nil, errors.Wrap(err, "could not retrieve randao signature set")
-	}
-	state, err = b.ProcessRandaoNoVerify(state, signed.Block.Body)
-	if err != nil {
-		traceutil.AnnotateError(span, err)
-		return nil, nil, errors.Wrap(err, "could not verify and process randao")
-	}
-
-	state, err = b.ProcessEth1DataInBlock(ctx, state, signed)
-	if err != nil {
-		traceutil.AnnotateError(span, err)
-		return nil, nil, errors.Wrap(err, "could not process eth1 data")
+	var bSet *bls.SignatureSet
+	var rSet *bls.SignatureSet
+	var aSet *bls.SignatureSet
+	var pipeline = []processFunc{
```

Comment:
Not the most elegant code, but I couldn't think of a better option to incorporate signature functions into the pipeline. To make the pipeline a package-level variable, it is necessary to move signature variables as well. I find this scoping clearer, but I don't have a strong opinion.